src/global/hooks/useGlobalAnalytics.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+
+import analyticsTracking from 'src/services/analytics-service';
+
+import { type AnalyticsOptions } from 'src/analyticsHelper';
+
+const useGlobalAnalytics = () => {
+  const location = useLocation();
+  const appName: string =
+    location.pathname.split('/').filter(Boolean)[0] || 'homepage';
```

Comment:
Don't we save our app names in redux? Why not read it back from there?

Reply:
Here is the error I get when I use redux:

```
No errors found.
TypeError: Cannot read properties of undefined (reading 'currentApp')
    at getCurrentApp (webpack://ensembl-new/./src/global/globalSelectors.ts?:24:45)
    at memoizedSelector (/Users/mapasa/Documents/work/ebi/repos/ensembl-client/node_modules/use-sync-external-store/cjs/use-sync-external-store-shim/with-selector.development.js:79:30)
    at getServerSnapshotWithSelector (/Users/mapasa/Documents/work/ebi/repos/ensembl-client/node_modules/use-sync-external-store/cjs/use-sync-external-store-shim/with-selector.development.js:138:14)
    at Object.useSyncExternalStore (/Users/mapasa/Documents/work/ebi/repos/ensembl-client/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:5221:10)
    at useSyncExternalStore (/Users/mapasa/Documents/work/ebi/repos/ensembl-client/node_modules/react/cjs/react.development.js:1676:21)
    at useSyncExternalStoreWithSelector (/Users/mapasa/Documents/work/ebi/repos/ensembl-client/node_modules/use-sync-external-store/cjs/use-sync-external-store-shim/with-selector.development.js:145:15)
    at useSelector (/Users/mapasa/Documents/work/ebi/repos/ensembl-client/node_modules/react-redux/lib/hooks/useSelector.js:54:27)
    at useGlobalAnalytics (webpack://ensembl-new/./src/global/hooks/useGlobalAnalytics.ts?:33:76)
    at eval (webpack://ensembl-new/./src/content/app/App.tsx?:68:82)
    at renderWithHooks (/Users/mapasa/Documents/work/ebi/repos/ensembl-client/node_modules/react-dom/cjs/react-dom-server-legacy.node.development.js:5662:16)
```